# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1422044589 18000
#      Fri Jan 23 15:23:09 2015 -0500
# Branch gui-release
# Node ID 844448ae53f3cd21653d6b56dfeb08a6ea9eba32
# Parent  709f3a77685d96d1264ca0e9a01d53d8398fa6f9
# Parent  75a671fcdd734eca88b4ca90a940bed45263924c
maint: Periodic merge of stable to gui-release.

diff --git a/liboctave/cruft/Faddeeva/Faddeeva.cc b/liboctave/cruft/Faddeeva/Faddeeva.cc
--- a/liboctave/cruft/Faddeeva/Faddeeva.cc
+++ b/liboctave/cruft/Faddeeva/Faddeeva.cc
@@ -134,16 +134,17 @@
                        For real x, use standard-library erf(x)
                        and erfc(x) if available (for C99 or C++11).
                        #include "config.h" if HAVE_CONFIG_H is #defined.
      15 December 2012: Portability fixes (copysign, Inf/NaN creation),
                        use CMPLX/__builtin_complex if available in C,
                        slight accuracy improvements to erf and dawson
                        functions near the origin.  Use gnulib functions
                        if GNULIB_NAMESPACE is defined.
+     18 December 2012: Slight tweaks (remove recomputation of x*x in Dawson)
 */
 
 /////////////////////////////////////////////////////////////////////////
 /* If this file is compiled as a part of a larger project,
    support using an autoconf-style config.h header file
    (with various "HAVE_*" #defines to indicate features)
    if HAVE_CONFIG_H is #defined (in GNU autotools style). */
 
@@ -205,16 +206,17 @@ static inline double my_copysign(double 
 // If we are using the gnulib <cmath> (e.g. in the GNU Octave sources),
 // gnulib generates a link warning if we use ::floor instead of gnulib::floor.
 // This warning is completely innocuous because the only difference between
 // gnulib::floor and the system ::floor (and only on ancient OSF systems)
 // has to do with floor(-0), which doesn't occur in the usage below, but
 // the Octave developers prefer that we silence the warning.
 #  ifdef GNULIB_NAMESPACE
 #    define floor GNULIB_NAMESPACE::floor
+#    define log GNULIB_NAMESPACE::log
 #  endif
 
 #else // !__cplusplus, i.e. pure C (requires C99 features)
 
 #  include "Faddeeva.h"
 
 #  define _GNU_SOURCE // enable GNU libc NAN extension if possible
 
@@ -579,17 +581,17 @@ cmplx FADDEEVA(Dawson)(cmplx z, double r
       return (1. / (-15 + x2*(90 + x2*(-60 + 8*x2)))) *
         C(x * (33 + x2 * (-28 + 4*x2)
                + y2 * (18 - 4*x2 + 4*y2)),
           y * (-15 + x2 * (24 - 4*x2)
                + y2 * (4*x2 - 10 - 4*y2)));
     }
     else {
       double D = spi2 * FADDEEVA(w_im)(x);
-      double x2 = x*x, y2 = y*y;
+      double y2 = y*y;
       return C
         (D + y2 * (D + x - 2*D*x2)
          + y2*y2 * (D * (0.5 - x2 * (2 - 0.66666666666666666667*x2))
                     + x * (0.83333333333333333333
                            - 0.33333333333333333333 * x2)),
          y * (1 - 2*D*x
               + y2 * 0.66666666666666666667 * (1 - x2 - D*x * (3 - 2*x2))
               + y2*y2 * (0.26666666666666666667 -
